src/main/kotlin/com/example/passmanager/repositories/impl/PassOwnerRepositoryImpl.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.passmanager.repositories.impl
+
+import com.example.passmanager.domain.MongoPassOwner
+import com.example.passmanager.repositories.PassOwnerRepository
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.findById
+import org.springframework.data.mongodb.core.query.Criteria.where
+import org.springframework.data.mongodb.core.query.Query.query
+import org.springframework.data.mongodb.core.query.isEqualTo
+import org.springframework.data.mongodb.core.remove
+import org.springframework.stereotype.Repository
+
+@Repository
+class PassOwnerRepositoryImpl(private val mongoTemplate: MongoTemplate) : PassOwnerRepository {
+    override fun findById(passOwnerId: String): MongoPassOwner? {
+        return mongoTemplate.findById(passOwnerId)
+    }
+
+    override fun insert(newMongoPassOwner: MongoPassOwner): MongoPassOwner {
+        return mongoTemplate.insert(newMongoPassOwner)
+    }
+
+    override fun deleteById(passOwnerId: String) {
+        mongoTemplate.remove<MongoPassOwner>(query(where("_id").isEqualTo(passOwnerId)))
```

Comment:
Does it work when pass String _id instead of ObjectId?

Reply:
Yes, mongo template recognizes "_id" field, and converts string to ObjectId

Logs:
> 2024-09-20T14:13:26.193+03:00 DEBUG 5779 --- [PassManagingApplication] [nio-8080-exec-3] o.s.data.mongodb.core.MongoTemplate      : Remove using query: { "passOwnerId" : { "$oid" : "66ed58c3cf2a271e42fd903a"}} in collection: pass.